tigervnc: 1.14.0 -> 1.15.0, also use `tag` / `hash` / `finalAttrs`

Yarny0 opened the pull request:
*   [switch from `rev` to `tag` in `fetchFromGitHub`](https://github.com/NixOS/nixpkgs/pull/355973)
*   replace `rec` with `finalAttrs`
*   update to [current release](https://github.com/TigerVNC/tigervnc/releases/tag/v1.15.0)

`tigervnc` has no maintainer, notifying previous non-treewide contributor @Artturin

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [Nixpkgs 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/doc/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/doc/manual/release-notes/rl-2505.section.md) Nixpkgs Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
- [NixOS 25.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2511.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) NixOS Release notes)
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

`nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"` tries to rebuild 5,800+ packages and was therefore skipped.

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Yarny0 committed:
tigervnc: use `tag` and `hash` in `fetchFromGitHub`

* `sha256` -> `hash`
* `rev` -> `tag`, inspired by
  https://github.com/NixOS/nixpkgs/commit/cb9f9a1e5a51a87d59b373db7016cee1608debd5

Yarny0 committed:
tigervnc: use `finalAttrs` pattern instead of `rec`

Yarny0 committed:
tigervnc: 1.14.0 -> 1.15.0

https://github.com/TigerVNC/tigervnc/releases/tag/v1.15.0

...also drop a patch that is now included in the release:
 https://github.com/TigerVNC/tigervnc/commit/4f6a3521874da5a67fd746389cfa9b6199eb3582

Defelo commented on pkgs/by-name/ti/tigervnc/package.nix:
```diff
@@ -25,25 +24,17 @@
-    rev = "v${version}";
-    sha256 = "sha256-TgVV/4MRsQHYKpDf9L5eHMLVpdwvNy1KPDIe7xMlQ9o=";
+    tag = "v${finalAttrs.version}";
+    sha256 = "sha256-ZuuvRJe/lAqULWObPxGHVJrDPCTK4IVSqX0K1rWOctw=";
```
```suggestion
    hash = "sha256-ZuuvRJe/lAqULWObPxGHVJrDPCTK4IVSqX0K1rWOctw=";
```

if we're already modernizing here

Yarny0 replied:
You're right, @Defelo !  Done and branch got updated.

Defelo commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 409035`

Logs: https://github.com/Defelo/nixpkgs-review-gha/actions/runs/15184595392

<details><summary>Download packages from cache:</summary><ul>
<li><details><summary><code>x86_64-linux</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-409035-x86_64-linux \
  --option binary-caches 'https://cache.nixos.org/ https://attic.defelo.de/nixpkgs' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  nixpkgs:xeaAWa3crK09hMmFiygBeRmLq3hUjUShgaAwYVUEtw0=
  ' \
  /nix/store/8vjg7h6rbjn7ybgqp2rcbxb0m5sf3nf9-crowbar-unstable-2020-04-23 \
  /nix/store/9xzyxhxm2ycqmsljrzic6kvqk28vqq94-crowbar-unstable-2020-04-23-dist \
  /nix/store/slsi48im6lmv1zpj47dcrmnv4gj44f7g-nemu-3.3.1 \
  /nix/store/csv57cvvqk41rh9yhircd7sxzbpip6d0-tigervnc-1.15.0
```
</details></li>
<li><details><summary><code>aarch64-linux</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-409035-aarch64-linux \
  --option binary-caches 'https://cache.nixos.org/ https://attic.defelo.de/nixpkgs' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  nixpkgs:xeaAWa3crK09hMmFiygBeRmLq3hUjUShgaAwYVUEtw0=
  ' \
  /nix/store/z690vmqprc1xkqi9yv8fn6aigbs911w5-crowbar-unstable-2020-04-23 \
  /nix/store/6fwcaa5knqmjic4dgp5w0xggpyxrk4p6-crowbar-unstable-2020-04-23-dist \
  /nix/store/4jmd7xnqj3sz9spg0hs0migllcnj9nfk-nemu-3.3.1 \
  /nix/store/5j6g27kg8bglc54wzrfx58kjp4j8p6ak-tigervnc-1.15.0
```
</details></li>
</ul></details>


---
### `x86_64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>crowbar</li>
    <li>crowbar.dist</li>
    <li>nemu</li>
    <li>tigervnc</li>
  </ul>
</details>


---
### `aarch64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>crowbar</li>
    <li>crowbar.dist</li>
    <li>nemu</li>
    <li>tigervnc</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = false)
<details>
  <summary>:x: 3 packages failed to build:</summary>
  <ul>
    <li>crowbar</li>
    <li>crowbar.dist</li>
    <li>tigervnc</li>
  </ul>
</details>


---
### `aarch64-darwin` (sandbox = false)
<details>
  <summary>:x: 3 packages failed to build:</summary>
  <ul>
    <li>crowbar</li>
    <li>crowbar.dist</li>
    <li>tigervnc</li>
  </ul>
</details>

Yarny0 committed:
tigervnc: broken on darwin

See also https://github.com/TigerVNC/tigervnc/issues/877 .

From reading

https://github.com/TigerVNC/tigervnc/issues/877

tigervnc is supposed to build on Darwin platform,
so we set the `broken` flag to indicate something's wrong here.

Defelo commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 409035`

Logs: https://github.com/Defelo/nixpkgs-review-gha/actions/runs/15213025612

<details><summary>Download packages from cache:</summary><ul>
<li><details><summary><code>x86_64-linux</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-409035-x86_64-linux \
  --option binary-caches 'https://cache.nixos.org/ https://attic.defelo.de/nixpkgs' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  nixpkgs:xeaAWa3crK09hMmFiygBeRmLq3hUjUShgaAwYVUEtw0=
  ' \
  /nix/store/8vjg7h6rbjn7ybgqp2rcbxb0m5sf3nf9-crowbar-unstable-2020-04-23 \
  /nix/store/9xzyxhxm2ycqmsljrzic6kvqk28vqq94-crowbar-unstable-2020-04-23-dist \
  /nix/store/slsi48im6lmv1zpj47dcrmnv4gj44f7g-nemu-3.3.1 \
  /nix/store/csv57cvvqk41rh9yhircd7sxzbpip6d0-tigervnc-1.15.0
```
</details></li>
<li><details><summary><code>aarch64-linux</code></summary>

```shell
nix-store -r --add-root nixpkgs-pr-409035-aarch64-linux \
  --option binary-caches 'https://cache.nixos.org/ https://attic.defelo.de/nixpkgs' \
  --option trusted-public-keys '
  cache.nixos.org-1:6NCHdD59X431o0gWypbMrAURkbJ16ZPMQFGspcDShjY=
  nixpkgs:xeaAWa3crK09hMmFiygBeRmLq3hUjUShgaAwYVUEtw0=
  ' \
  /nix/store/z690vmqprc1xkqi9yv8fn6aigbs911w5-crowbar-unstable-2020-04-23 \
  /nix/store/6fwcaa5knqmjic4dgp5w0xggpyxrk4p6-crowbar-unstable-2020-04-23-dist \
  /nix/store/4jmd7xnqj3sz9spg0hs0migllcnj9nfk-nemu-3.3.1 \
  /nix/store/5j6g27kg8bglc54wzrfx58kjp4j8p6ak-tigervnc-1.15.0
```
</details></li>
</ul></details>


---
### `x86_64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>crowbar</li>
    <li>crowbar.dist</li>
    <li>nemu</li>
    <li>tigervnc</li>
  </ul>
</details>


---
### `aarch64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>crowbar</li>
    <li>crowbar.dist</li>
    <li>nemu</li>
    <li>tigervnc</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:fast_forward: 3 packages marked as broken and skipped:</summary>
  <ul>
    <li>crowbar</li>
    <li>crowbar.dist</li>
    <li>tigervnc</li>
  </ul>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:fast_forward: 3 packages marked as broken and skipped:</summary>
  <ul>
    <li>crowbar</li>
    <li>crowbar.dist</li>
    <li>tigervnc</li>
  </ul>
</details>
